app/models/spree/gateway/stripe_connect.rb

```diff
@@ -0,0 +1,152 @@
+module Spree
+  class Gateway
+    class StripeConnect < Gateway
+      preference :enterprise_id, :integer
+
+      attr_accessible :preferred_enterprise_id
+
+      CARD_TYPE_MAPPING = {
+        'American Express' => 'american_express',
+        'Diners Club' => 'diners_club',
+        'Visa' => 'visa'
+      }.freeze
+
+      def method_type
+        'stripe'
+      end
+
+      def provider_class
+        ActiveMerchant::Billing::StripeGateway
+      end
+
+      def payment_profiles_supported?
+        true
+      end
+
+      def stripe_account_id
+        StripeAccount.find_by_enterprise_id(preferred_enterprise_id).andand.stripe_user_id
+      end
+
+      def purchase(money, creditcard, gateway_options)
+        provider.purchase(*options_for_purchase_or_auth(money, creditcard, gateway_options))
+      end
+
+      # def authorize(money, creditcard, gateway_options)
+      #   provider.authorize(*options_for_purchase_or_auth(money, creditcard, gateway_options))
+      # end
+
+      # def capture(money, response_code, gateway_options)
+      #   provider.capture(money, response_code, gateway_options)
+      # end
+
+      # def credit(money, creditcard, response_code, gateway_options)
+      #   provider.refund(money, response_code, {})
+      # end
+
+      # def void(response_code, creditcard, gateway_options)
+      #   provider.void(response_code, {})
+      # end
+
+      # def cancel(response_code)
+      #   provider.void(response_code, {})
+      # end
+
+      def create_profile(payment)
+        return unless payment.source.gateway_customer_profile_id.nil?
+        options = {
+          email: payment.order.email,
+          login: Stripe.api_key,
+        }.merge! address_for(payment)
```

Comment:
Not super important, but if you move this hash population to a separate method you'll have more documentation surface and we'll get prepared for the rubocop rules about methods and classes length I hope we'll agree upon some day 😬

Reply:
Resolved